frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { PipelineParam } from '@console/dev-console/src/utils/pipeline-augment';
+import { pipelineParameters, pipelineParametersWithoutDefaults } from './utils-data';
+import { removeEmptyDefaultFromPipelineParams } from '../utils';
+
+describe('removeEmptyDefaultFromPipelineParams omits empty default values', () => {
+  it('should return pipline parameters by only omitting empty default values', () => {
+    const result = removeEmptyDefaultFromPipelineParams(pipelineParameters);
+    const expectedPipelineParameters: PipelineParam[] = [
+      {
+        name: 'param1',
+        default: 'abc',
+        description: 'This is param 1',
+      },
+      {
+        name: 'param2',
+        description: 'This is param 2',
+      },
+      {
+        name: 'param3',
+        default: 'xyz',
+        description: 'This is param 3',
+      },
+    ];
+
+    expect(result).toEqual(expectedPipelineParameters);
+  });
+
+  it('should return empty array if pipline parameters is empty', () => {
+    let result = removeEmptyDefaultFromPipelineParams(null);
+    expect(result).toEqual([]);
+
+    result = removeEmptyDefaultFromPipelineParams([]);
+    expect(result).toEqual([]);
+  });
+
+  it('should return pipline parameters as is if default is non-empty', () => {
+    pipelineParameters[1].default = 'mno';
```

Comment:
Probably want to avoid direct modifications. Object references come through imports and could impact other tests that want to use the same data - which usually rears its head at "random" depending on which order the tests are executed in. It's possible Jest protects us from this but I'd be surprised if they do.

Just use lodash to deep clone before modifying directly. Safety first 😄